bubble_sort.rb

```diff
@@ -0,0 +1,36 @@
+def bubble_sort(arr)
+  i = 0
+  while i < arr.length - 1
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i] > arr[i + 1]
+    if i == arr.length - 2
+      break unless arr != arr.sort
+
+      i = 0
+      next
+    end
+    i += 1
+  end
+  puts arr.to_s
+end
+
+bubble_sort([4, 3, 78, 2, 0, 2])
+
+def bubble_sort_by(arr)
+  i = 0
+  while i < arr.length - 1
+    next unless yield(arr[i], arr[i + 1])
+
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i].length > arr[i + 1].length
+    if i == arr.length - 2
+      break unless arr != arr.sort { |x, y| x.length <=> y.length }
+
+      i = 0
+    end
+    i += 1
+  end
+  puts arr.to_s
+end
+
+bubble_sort_by(%w[hello hi hey]) do |first, second|
+  first.length <=> second.length
+end
```

Comment:
Remove your test example.

Reply:
The method uses a yield removing it cant break the yield. I am not sure I understand what you meant by removing the example here.

Reply:
You have tested your bubble_sort_by on the ['hello', 'hi', 'hey'] enum, right?